lib/media/media_source_engine.js

```diff
@@ -696,11 +696,28 @@ shaka.media.MediaSourceEngine = class {
               goog.asserts.assert(
                   box.version == 0 || box.version == 1,
                   'TFDT version can only be 0 or 1');
-
-              const parsedTFDTBox = shaka.util.Mp4BoxParsers.parseTFDT(
-                  box.reader, box.version);
-              const baseTime = parsedTFDTBox.baseMediaDecodeTime;
-              startTime = baseTime / timescale;
+              const Mp4BoxParsers = shaka.util.Mp4BoxParsers;
+              try {
+                const parsedTFDTBox = Mp4BoxParsers.parseTFDT(
+                    box.reader, box.version);
+                const baseTime = parsedTFDTBox.baseMediaDecodeTime;
+                startTime = baseTime / timescale;
+              } catch (error) {
+                if (error.code != shaka.util.Error.Code.JS_INTEGER_OVERFLOW) {
+                  throw error;
+                }
+                if (!window.BigInt) {
+                  throw error;
+                }
+                const parsedTFDTBox = Mp4BoxParsers.parseTFDTAsBigInt(
+                    box.reader);
+                const baseTime = parsedTFDTBox.baseMediaDecodeTime;
+                const startTimeBigInt = baseTime / BigInt(timescale);
+                if (startTimeBigInt > Number.MAX_SAFE_INTEGER) {
+                  throw error;
+                }
+                startTime = Number(startTimeBigInt);
```

Comment:
Because this output is in seconds, lost precision in the BigInt parsing would not be noticed.  I'm okay with us getting an imprecise value here, because it's merely in seconds, and a few microseconds difference is not going to hurt anything.  But the parser itself looks flawed, and someone using it to try to get a precise baseMediaDecodeTime for other purposes might have issues.

Since the output here does not need to be precise, why don't we do this without a BigInt dependency at all?

Reply:
There are some operations that cannot be done when bigint and number are present (see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/Cant_convert_BigInt_to_number ) That's why I searched for this solution.

Reply:
BigInt isn't available everywhere (smart TVs).  So since we don't need accuracy in this case, let's just be inaccurate.  No dependencies on BigInt required.

```js
  /**
   * Parses a TFDT Box, with a loss of precision beyond 53 bits.
   * Use only when exact integers are not required, e.g. when dividing by the timescale.
   *
   * @param {!shaka.util.DataViewReader} reader
   * @param {number} version
   * @return {!shaka.util.ParsedTFDTBox}
   */
  static parseTFDTInaccurate(reader, version) {
    if (version == 1) {
      const high = reader.readUint32();
      const low = reader.readUint32();
      return {
        baseMediaDecodeTime: (high * Math.pow(2, 32)) + low,
      };
    } else {
      return {
        baseMediaDecodeTime: reader.readUint32(),
      };
    }
  }

  const Mp4BoxParsers = shaka.util.Mp4BoxParsers;
  const parsed = Mp4BoxParsers.parseTFDTInaccurate(box.reader, box.version);
  startTime = parsed.baseMediaDecodeTime / timescale;
```

Reply:
OK, I changed the implementation